catalog/resourcemanagement/resourcemanagement-usage-plugin/src/main/java/org/codice/ddf/resourcemanagement/usage/ResourceUsagePlugin.java

```diff
@@ -62,15 +64,60 @@ public ResourceRequest process(ResourceRequest input)
 
                     String username = getUsernameFromSubject(input.getPropertyValue(
                             SecurityConstants.SECURITY_SUBJECT));
+                    if (StringUtils.isNotEmpty(username)) {
+                        long currentUserDataUage = 0L;
+                        long userDataLimit = 0L;
+
+                        try {
+                            currentUserDataUage =
+                                    attributesStore.getCurrentDataUsageByUser(username);
+                            userDataLimit = attributesStore.getDataLimitByUser(username);
+                        } catch (PersistenceException pex) {
+                            LOGGER.warn("Persistence exception updating user {} data usage",
+                                    username,
+                                    pex);
+                        }
+                        if (currentUserDataUage + resourceSize > userDataLimit) {
+                            throw new StopProcessingException(
+                                    " " + username + ": data usage limit exceeded. (" + userDataLimit
+                                            + " bytes)");
+                        }
+                    }
+                }
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public ResourceResponse process(ResourceResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        long resourceSize = 0L;
+
+        if (input != null) {
+
+            Object sizeObj = input.getPropertyValue(Metacard.RESOURCE_SIZE);
+            if (sizeObj != null && sizeObj instanceof String) {
```

Comment:
You can just do `sizeObj instanceof String` since `instanceof` is always false for null.